MatrixSDK/Threads/MXThreadingService.swift

```diff
@@ -0,0 +1,142 @@
+// 
+// Copyright 2021 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// MXThreadingService error
+public enum MXThreadingServiceError: Int, Error {
+    case sessionNotFound
+    case unknown
+}
+
+// MARK: - MXThreadingService errors
+extension MXThreadingServiceError: CustomNSError {
+    public static let errorDomain = "org.matrix.sdk.threadingservice"
+
+    public var errorCode: Int {
+        return Int(rawValue)
+    }
+
+    public var errorUserInfo: [String: Any] {
+        return [:]
+    }
+}
+
+@objcMembers
+/// Threading service class.
+public class MXThreadingService: NSObject {
+    
+    private weak var session: MXSession?
+    
+    private var threads: [String: MXThread] = [:]
+    
+    /// Notification to be posted when a new thread is created.
+    public static let newThreadCreated: Notification.Name = Notification.Name("MXThreadingService.newThreadCreated")
+    
+    /// Initializer
+    /// - Parameter session: session instance
+    public init(withSession session: MXSession) {
+        self.session = session
+        super.init()
+    }
+    
+    /// Adds event to the related thread instance
+    /// - Parameter event: event to be handled
+    public func handleEvent(_ event: MXEvent) {
+        guard let session = session else {
+            //  session closed
+            return
+        }
+        guard let threadIdentifier = event.threadIdentifier else {
```

Comment:
Given the use of `threadId` elsewhere do you think it would make sense for the property on `MXEvent` to match this name too?
Edit: Seems like this was part of a previous commit, I didn't realise. Maybe out of scope?

Reply:
Yeah makes perfect sense, updated.